recipes/full_finetune_distributed.py

```diff
@@ -440,6 +460,26 @@ def _is_layer_fqn(s: str) -> bool:
             cpu_offload=fsdp_cpu_offload,
         )
 
+        # activation checkpointing/offloading
+        if enable_activation_offloading:
+            if self._device.type != "cuda":
+                raise RuntimeError(
+                    "enable_activation_offloading should only be True for training on CUDA"
+                )
+            if not enable_activation_checkpointing:
+                raise RuntimeError(
+                    "enable_activation_offloading should only be True when enable_activation_checkpointing is True"
+                )
+        elif enable_activation_checkpointing:
+            log.info(
+                "Hint: enable_activation_checkpointing is True, but enable_activation_offloading isn't. "
+                "Enabling activation offloading should reduce memory further."
+            )
```

Comment:
the only think that i dont like about this is that it could fail much faster if we added it to the __init__. But i like that the checks are near the code where it matters. So i am not sure which one to pick.

Reply:
I think failing much faster is better